provider/resource_corp_integration.go

```diff
@@ -22,6 +23,12 @@ func resourceCorpIntegration() *schema.Resource {
 				Description: "One of (mailingList, slack, microsoftTeams)",
 				Required:    true,
 				ForceNew:    true,
+				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+					if existsInString(val.(string), "mailingList", "slack", "microsoftTeams") {
+						return nil, nil
+					}
+					return nil, []error{fmt.Errorf("received type '%s' is invalid. should be 'mailingList', 'slack', or 'microsoftTeams'", val.(string))}
+				},
```

Comment:
@ponkio-o In my experience, it's more common in Go to check the error case first:

```suggestion
				ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
					if !existsInString(val.(string), "mailingList", "slack", "microsoftTeams") {
						return nil, []error{fmt.Errorf("received type %q is invalid. should be "mailingList", "slack", or "microsoftTeams"`, val.(string))}
					}
					return nil, nil
				},
```

At some point I can update the rest of the validate functions to match this as well.

Reply:
👀 
Thank you for teaching me! I learned something new.